src/lib/fs.c

```diff
@@ -0,0 +1,63 @@
+#include <ftw.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "fs.h"
+#include "../tracing.h"
+
+int DirEnsure(const char *path)
+{
+	int rv;
+	struct stat st = {0};
+
+	rv = stat(path, &st);
+	if (rv == 0) {
+		if (!S_ISDIR(st.st_mode)) {
+			tracef("%s is not a directory", path);
+			return -1;
+		}
+	}
+
+	/* Directory does not exist */
+	if (rv == -1) {
+		return mkdir(path, 0700);
+	}
+
+	return 0;
+}
+
+
+static int dir_tear_down_nftw_fn(const char *       path,
```

Comment:
Nit: naming convention for local unexported functions should be `<moduleName><FunctionName>`, in this case perhaps something like `fsRemoveDirFilesNftwFn`.